sql/src/main/java/io/crate/exceptions/SQLExceptions.java

```diff
@@ -124,7 +124,7 @@ public static SQLActionException createSQLActionException(Throwable e, Exception
         e = esToCrateException(e);
         try {
             validator.ensureExceptionAuthorized(e);
-        } catch (MissingPrivilegeException mpe) {
+        } catch (Exception mpe) {
```

Comment:
Couldn't this potentially leak unwanted exceptions from the validator?

Reply:
Don't think so. All exceptions the validator throws are things the user should get. If there is a programming error within the validator we'll want to catch that as well - given that `createSQLActionException` is likely to be called in onException handlers and not guarded against other exceptions, so raising here something could cause operations to get stuck

Reply:
I think the transform method would be a cleaner way. It would help us to differentiate between transformed exceptions (i.e. privilege checked exceptions) as return value and other exceptions as regular exceptions.